house/people.go

```diff
@@ -3,54 +3,77 @@ package house
 import (
 	"context"
 	"fmt"
-	"github.com/google/uuid"
-	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/client_golang/prometheus/promauto"
+	pb "github.com/beaujr/nmap_prometheus/proto"
+	etcdv3 "github.com/ozonru/etcd/v3/clientv3"
 	"gopkg.in/yaml.v2"
 	"log"
-	"time"
 )
 
-type people struct {
-	Id      string   `json:"id",yaml:"id"`
-	Devices []string `json:"devices",yaml:"devices"`
-	Name    string   `json:"name",yaml:"name"`
+var peoplePrefix = "/people/"
+
+// Notifier represents and interface for notification sending
+type PeopleManager interface {
+	Get() ([]*pb.Person, error)
+	CreateFromDevices(devices []*pb.Devices, name string) error
+	Create(ids []string, name string) error
+}
+
+// NewNotifier returns a new Notifier
+func NewPeopleManager(etcdClient etcdv3.KV) PeopleManager {
+	return &EtcdPeopleManager{etcdClient: etcdClient}
 }
 
-func (s *Server) createPerson(devices []string, name string) error {
-	// Create a queue
-	person := &people{
-		Name:    name,
-		Devices: devices,
-		Id:      uuid.New().String(),
+// FCMNotifier is an implementation of the Notifier
+type EtcdPeopleManager struct {
+	Notifier
+	etcdClient etcdv3.KV
+}
+
+func (etm *EtcdPeopleManager) Get() ([]*pb.Person, error) {
+	items, err := etm.etcdClient.Get(context.Background(), fmt.Sprintf("%s", peoplePrefix), etcdv3.WithPrefix())
+	people := make([]*pb.Person, 0)
+	if err != nil {
+		return nil, err
 	}
-	metricId := fmt.Sprintf("%s%s", metricsKey, person.Id)
-	if metrics[metricId] == nil {
-		metrics[metricId] = promauto.NewGauge(prometheus.GaugeOpts{
-			Name: "home_detector_people_devices",
-			Help: "BleDevice in home",
-			ConstLabels: prometheus.Labels{
-				"name": name,
-				"id":   person.Id,
-			},
-		})
+	if items == nil {
+		return people, nil
 	}
-	metrics[metricId].Set(float64(time.Now().Unix()))
-	err := s.writePerson(person)
-	if err != nil {
-		log.Println(err)
+
+	for _, person := range items.Kvs {
+		var dev *pb.Person
+		err = yaml.Unmarshal(person.Value, &dev)
+		if err != nil {
+			return nil, err
+		}
+		people = append(people, dev)
+	}
+	return people, nil
+}
+
+func (etm *EtcdPeopleManager) CreateFromDevices(devices []*pb.Devices, name string) error {
+	ids := make([]string, 0)
+	for _, item := range devices {
+		ids = append(ids, item.Id.Mac)
 	}
+	return etm.Create(ids, name)
 
-	return nil
 }
 
-func (s *Server) writePerson(person *people) error {
+func (etm *EtcdPeopleManager) Create(ids []string, name string) error {
```

Comment:
warning: exported method EtcdPeopleManager.Create should have comment or be unexported (golint)